### ISSUE TITLE
not working (windows)

### ISSUE THREAD

**Issue by CapivaraTupiniquim** [WillNilges/inkpath#33]
I probably don't know what I'm doing, sorry, but it doesn't work (windows)

![image](https://user-images.githubusercontent.com/67530921/214759106-241f520f-08f4-4994-b050-35eba1848063.png)

**Comment by WillNilges** [comment]
Hey there! Thanks for opening an issue!

I haven't actually put any work into getting this working on Windows yet, but this might be the kick I need to see if I can. Now that it's ported to OpenCV, it should be possible.

This app is very much not yet finished and most of my efforts have gone into optimizing the Lua code (and helping Xournal++ release v1.2, which has API calls needed for this plugin to work). 

If you'd like to contribute, you're more than welcome to! You'll need to build Xournal++ from source, and have a statically linked version of OpenCV 4.5 installed. You can find instructions for building Xournalpp linked in the README, and if you give me some time, I can probably dig up the instructions I used to compile OpenCV, or you can look at the Dockerfiles and related scripts in `HACKING/`.

**Comment by GabrielOsvair** [comment]
Got it! No problems. I can't help with the development (other than reporting bugs) as I don't know how to do that, but I wish you all the best.

**Comment by MiltonBalaOfficial** [comment]
will it be available for windows soon?

**Comment by LucaSpallanzani** [comment]
> will it be available for windows soon?

?? :)

**Comment by WillNilges** [comment]
Hey there, thanks for your interest! I'm starting to pick up development on this again, and I have plans to build a version for Windows and macOS. No promises on when that will happen, but I would like to get something out the door sooner rather than later.

I have no experience packaging things for Windows, but I might be able to at least provide build instructions. Would that suffice?

**Comment by LucaSpallanzani** [comment]
> will it be available for windows soon?



> Hey there, thanks for your interest! I'm starting to pick up development on this again, and I have plans to build a version for Windows and macOS. No promises on when that will happen, but I would like to get something out the door sooner rather than later.
> 
> I have no experience packaging things for Windows, but I might be able to at least provide build instructions. Would that suffice?

Yeah, for sure! But no need to rush, just wanted to manifest my interest on your plug-in, hope to be using it soon, thanks!!

**Comment by WillNilges** [comment]
Alright, it took some wrestling with DLLs, but I think I'm close to having a usable Windows build. Inkpath seems to work, so now just a matter of packaging it nicely.

**Comment by MiltonBalaOfficial** [comment]
> Alright, it took some wrestling with DLLs, but I think I'm close to having a usable Windows build. Inkpath seems to work, so now just a matter of packaging it nicely.

great news!

**Comment by WillNilges** [comment]
Alright folks, I've finished packaging a preliminary build for this app. Essentially, I used CMAKE to cross compile on mingw into a dll, and then bundled that in the ImageTranscription directory with all the dependencies required for the plugin to work. I included a script to update the user's path to make that work.

I updated the release at https://github.com/WillNilges/inkpath/releases/tag/v0.2.0 to include ImageTranscription-win.zip. If you could download that, read the README instructions (also found [here](https://github.com/WillNilges/inkpath/blob/wdn/windows/README.md#installation-on-windows)), I would love some feedback. I'll warn you that the program itself is nowhere near complete. The transcription algorithm still needs a lot of work, which I intend to work on in the coming weeks.

If you're willing to help me out, I'd appreciate some sample images if you have any. Please include screenshots of Inkpath's output with it if you can.

**Comment by MiltonBalaOfficial** [comment]
https://github.com/WillNilges/inkpath/issues/36#issue-2754508626, someone help me with detailed instruction. I cannot make it work. Thanks in advance.

**Comment by WillNilges** [comment]
I've closed the issue now that I have a solid set of build instructions and released a Windows version. Please let me know if you run into issues.
